src/System.Management.Automation/engine/parser/CharTraits.cs

```diff
@@ -234,180 +234,92 @@ static CharExtensions()
+            || c == SpecialChars.QuoteLowDoubleLeft;
 
         // Return true if the character can be the first character of
         // the variable name for unbraced variable tokens.
-        internal static bool IsVariableStart(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.VarNameFirst) != 0;
-            }
-
-            return char.IsLetterOrDigit(c);
-        }
+        internal static bool IsVariableStart(this char c) =>
+            (c < 128 && (s_traits[c] & CharTraits.VarNameFirst) != 0)
+            || char.IsLetterOrDigit(c);
 
         // Return true if the character can be the first character of
         // an identifier or label.
-        internal static bool IsIdentifierStart(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.IdentifierStart) != 0;
-            }
-
-            return char.IsLetter(c);
-        }
+        internal static bool IsIdentifierStart(this char c) =>
+            (c < 128 && (s_traits[c] & CharTraits.IdentifierStart) != 0)
+            || char.IsLetter(c);
 
         // Return true if the character can follow the first character of
         // an identifier or label.
-        internal static bool IsIdentifierFollow(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & (CharTraits.IdentifierStart | CharTraits.Digit)) != 0;
-            }
-
-            return char.IsLetterOrDigit(c);
-        }
+        internal static bool IsIdentifierFollow(this char c) =>
+            (c < 128 && (s_traits[c] & (CharTraits.IdentifierStart | CharTraits.Digit)) != 0)
+            || char.IsLetterOrDigit(c);
 
         // Return true if the character is a hexadecimal digit.
-        internal static bool IsHexDigit(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.HexDigit) != 0;
-            }
-
-            return false;
-        }
+        internal static bool IsHexDigit(this char c) =>
+            c < 128 && (s_traits[c] & CharTraits.HexDigit) != 0;
 
         // Return true if the character is a decimal digit.
-        internal static bool IsDecimalDigit(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.Digit) != 0;
-            }
-
-            return false;
-        }
+        internal static bool IsDecimalDigit(this char c) =>
+            c < 128 && (s_traits[c] & CharTraits.Digit) != 0;
 
         // Return true if the character is a binary digit.
-        internal static bool IsBinaryDigit(this char c)
-        {
-            if (c < 128)
-            {
-                return (s_traits[c] & CharTraits.BinaryDigit) != 0;
-            }
-
-            return false;
-        }
+        internal static bool IsBinaryDigit(this char c) =>
+            c < 128 && (s_traits[c] & CharTraits.BinaryDigit) != 0;
```

Comment:
Oh, I meant that we could use CoreFX pattern for IsDecimalDigit and IsBinaryDigit to get a bit perf win.
Please remove commit.

Reply:
All good! Sorry, I guess I misinterpreted what you meant. 🙂

Reply:
@iSazonov is there a significant perf difference using the CoreFX pattern?

Reply:
The pattern is very fast. In common for whole PowerShell it is a bit of cause.  But you could measure exactly ScanNumber() and probably see perf win in some %. 
You can ignore this if you are tired.

Reply:
Unused code should be removed.